nix/modules/nixos/holo-orchestrator.nix

```
@@ -0,0 +1,177 @@
+  ...
+}:
+
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.holo.orchestrator;
+in
+{
+  imports = [
+    inputs.extra-container.nixosModules.default
+  ];
+
+  options.holo.orchestrator = {
+    enable = lib.mkOption {
+      description = "enable holo-orchestrator";
+      default = true;
+    };
+
+    autoStart = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = inputs.self.packages.${pkgs.stdenv.system}.rust-workspace;
+    };
+
+    hubAuthScriptPath =  lib.mkOption {
+      type = lib.types.path;
+      default = "${cfg.package}/../../../scripts/hub_auth_setup";
+    };
+
+    rust = {
+      log = lib.mkOption {
+        type = lib.types.str;
+        default = "debug";
+      };
+
+      backtrace = lib.mkOption {
+        type = lib.types.str;
+        default = "1";
+      };
+    };
+
+    mongo = {
+      documentation.enable = false;
+
+      listenHost = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+
+      listenPort = lib.mkOption {
+        type = lib.types.int;
+        default = 27017;
+      };
+
+      url = lib.mkOption {
+        type = lib.types.str;
+        default = "${cfg.mongo.listenHost}:${builtins.toString cfg.mongo.listenPort}";
+      };
+
+      services.mongodb = {
+        enable = true;
+        bind_ip = cfg.mongo.listenHost;
+      };
+
+      systemd.services.mongodb.serviceConfig = {
+        LimitNOFILE = 500000;
+      };
+
+      networking.firewall.allowedTCPPorts = [ cfg.mongo.listenPort ];
+    };
+
+    nats = {
+      listenHost = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+
+      listenPort = lib.mkOption {
+        type = lib.types.int;
+        default = 4222;
+      };
+
+      url = lib.mkOption {
+        type = lib.types.str;
+        default = "${cfg.nats.listenHost}:${builtins.toString cfg.nats.listenPort}";
+      };
+
+      tlsInsecure = lib.mkOption {
+        type = lib.types.bool;
+      };
+
+      nscPath = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/.local/share/nats/nsc";
+      };
+
+      sharedCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/shared_creds";
+      };
+
+      localCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/local_creds";
+      };
+
+      rootAuthNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/AUTH_ROOT_SK.nk";
+      };
+
+      signingAuthNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/AUTH_SK.nk";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.holo-orchestrator = {
+      enable = true;
+
+      after = [
+        "network.target"
+        "network-online.target"
```

one of my recent leanings is that we only need one of these. i guess we only need the network interfaces to be up, hence:

```suggestion
        "network.target"
```